include/power_grid_model/math_solver/sparse_lu_solver.hpp

```diff
@@ -90,18 +90,19 @@ class SparseLUSolver {
     }
 
     // solve with new matrix data, need to factorize first
-    void solve(std::vector<Tensor>& data,         // matrix data, factorize in-place
-               BlockPermArray& block_perm_array,  // pre-allocated permutation array, will be overwritten
-               std::vector<RHSVector> const& rhs, std::vector<XVector>& x) {
+    void prefactorize_and_solve(
+        std::vector<Tensor>& data,         // matrix data, factorize in-place
+        BlockPermArray& block_perm_array,  // pre-allocated permutation array, will be overwritten
+        std::vector<RHSVector> const& rhs, std::vector<XVector>& x) {
         prefactorize(data, block_perm_array);
         // call solve with const method
-        solve((std::vector<Tensor> const&)data, block_perm_array, rhs, x);
+        solve_with_prefactorization((std::vector<Tensor> const&)data, block_perm_array, rhs, x);
     }
 
     // solve with existing pre-factorization
-    void solve(std::vector<Tensor> const& data,         // pre-factoirzed data, const ref
-               BlockPermArray const& block_perm_array,  // pre-calculated permutation, const ref
-               std::vector<RHSVector> const& rhs, std::vector<XVector>& x) {
+    void solve_with_prefactorization(std::vector<Tensor> const& data,         // pre-factoirzed data, const ref
```

Comment:
Maybe call this one `solve_prefactorised`? solve with prefactorization sounds like the prefactorization also happens in this function